presto-main/src/main/java/io/prestosql/server/SessionPropertyDefaults.java

```diff
@@ -106,8 +116,34 @@ public Session newSessionWithDefaultProperties(Session session, Optional<String>
                 queryType,
                 resourceGroupId);
 
-        Map<String, String> systemPropertyOverrides = configurationManager.getSystemSessionProperties(context);
+        Map<String, String> systemPropertyOverrides = filterInvalidSystemPropertyOverrides(configurationManager.getSystemSessionProperties(context));
         Map<String, Map<String, String>> catalogPropertyOverrides = configurationManager.getCatalogSessionProperties(context);
         return session.withDefaultProperties(systemPropertyOverrides, catalogPropertyOverrides);
     }
+
+    private Map<String, String> filterInvalidSystemPropertyOverrides(Map<String, String> systemProperties)
+    {
+        ImmutableMap.Builder<String, String> validProperties = ImmutableMap.builder();
+
+        for (Map.Entry<String, String> property : systemProperties.entrySet()) {
+            try {
+                sessionPropertyManager.validateSystemSessionProperty(property.getKey(), property.getValue());
+            }
+            catch (RuntimeException e) {
+                log.error(e, format("Failed to set value %s for property %s",
```

Comment:
Place each argument on a separate line when splitting across multiple lines or keep them all on the same line.